core/src/main/java/it/pagopa/selfcare/onboarding/core/InstitutionServiceImpl.java

```diff
@@ -118,13 +118,25 @@ public void onboardingPaAggregator(OnboardingData onboardingData) {
     }
 
     @Override
-    public void onboardingCompanyV2(OnboardingData onboardingData) {
+    public void onboardingCompanyV2(OnboardingData onboardingData, String userFiscalCode) {
         log.trace("onboardingProductAsync start");
         log.debug("onboardingProductAsync onboardingData = {}", onboardingData);
+        InstitutionInfoIC userBusinesses = partyRegistryProxyConnector.getInstitutionsByUserFiscalCode(userFiscalCode);
+        if(businessIsNotRelatedToUser(userBusinesses, onboardingData.getTaxCode())){
+            throw new OnboardingNotAllowedException("The selected business does not belong to the user");
+        }
         onboardingMsConnector.onboardingCompany(onboardingData);
         log.trace("onboarding end");
     }
 
+    private boolean businessIsNotRelatedToUser(InstitutionInfoIC institutionInfoIC, String businessTaxCode) {
```

Comment:
The method businessIsNotRelatedToUser uses a negative naming convention, which can sometimes lead to confusion or double negatives in logic.

It might be better to Rename the method to a positive assertion for better readability and maintainability. For example:

```
private boolean isBusinessRelatedToUser(InstitutionInfoIC institutionInfoIC, String businessTaxCode) {
    return institutionInfoIC != null
            && CollectionUtils.isNotEmpty(institutionInfoIC.getBusinesses())
            && institutionInfoIC.getBusinesses()
               .stream()
               .anyMatch(business -> Objects.equals(business.getBusinessTaxId(), businessTaxCode));
}
```